gunicorn/workers/base.py

```diff
@@ -119,7 +119,13 @@ def changed(fname):
                 time.sleep(0.1)
                 sys.exit(0)
 
-            self.reloader = preferred_reloader(callback=changed)
+            reloader_cls = reloader_engines.get(self.cfg.reload_engine)
+
+            if not reloader_cls:
+                raise ValueError("Invalid reload_engine: %s"
```

Comment:
Oh, and also, wouldn't it be better to move this validation into `ReloadEngine.validatior`? This way we could easily add some unit tests.